pkg/getter/plugingetter.go

```diff
@@ -79,6 +79,10 @@ func (p *pluginGetter) Get(href string) (*bytes.Buffer, error) {
 	return buf, nil
 }
 
+func (p *pluginGetter) GetWithCredentials(href, username, password string) (*bytes.Buffer, error) {
+	return p.Get(href)
```

Comment:
This is the reason for the above comment... This is a very odd design choice. Why should Getters provide a `GetWithCredentials` function if they aren't going to use the credentials provided?

Reply:
@bacongobbler,
If understand your comment correctly, you're saying that only the *getter.httpGetter* struct actually needs credentials.  This will probably require the following the change:
The *ChartRepository* struct currently holds a *getter.Getter*. We'll need to replace it with *getter.httpGetter*. We need this because the *ChartDownloader.DownloadTo* method uses the getter from the *ChartRepository* struct, to download with the credentials.
Does this makes sense? Do you think it is okay to have the *ChartRepository* hold *getter.httpGetter* instead of a generic *getter.Getter*?
If so, I'll make this change.

Reply:
I do not think that it's okay to change the ChartRepository to use a `getter.HTTPGetter`. My current thinking is that when `ResolveChartVersionAndGetRepo` is called, it should be returning a `getter.HTTPGetter` with the username and password configuration as part of the struct, essentially moving [this entire code of logic](https://github.com/eyalbe4/helm/blob/50a3306a72830e1308c37437866d7abee0173d09/pkg/downloader/chart_downloader.go#L93-L104) into `ResolveChartVersionAndGetRepo`.

Does that help make sense?

Reply:
Sure @bacongobbler. Done.